From c448b6c175fbd82dd1ae1ffbbf34e47a8e7f9b2a Mon Sep 17 00:00:00 2001
From: Jan Provaznik <jprovazn@redhat.com>
Date: Mon, 14 Apr 2014 10:22:54 -0400
Subject: [PATCH] Allow install mariadb from RDO repository

Adds elements which allow installing mariadb from RDO repository.
It's similar to mariadb element but different repo with different
package names are used (also some installation steps might be removed
in future).

Common code for mariadb and mariadb-rdo is now in mariadb-common element.

Change-Id: I8566b03a24fff7984f323442d93249e1002a6ebe
---
 elements/mariadb-common/README.md                  |  2 ++
 elements/mariadb-common/element-deps               |  1 +
 elements/mariadb-common/install.d/11-mariadb       | 27 ++++++++++++++++
 .../os-refresh-config/post-configure.d/40-mariadb  |  4 +++
 .../pre-configure.d/50-mariadb-socket              | 10 ++++++
 elements/mariadb-dev-rdo/README.md                 |  4 +++
 elements/mariadb-dev-rdo/element-deps              |  1 +
 elements/mariadb-dev-rdo/element-provides          |  1 +
 elements/mariadb-dev-rdo/install.d/03-mariadb-dev  |  5 +++
 elements/mariadb-rdo/README.md                     |  4 +++
 elements/mariadb-rdo/element-deps                  |  2 ++
 elements/mariadb-rdo/element-provides              |  1 +
 elements/mariadb-rdo/install.d/10-mariadb-packages | 13 ++++++++
 elements/mariadb/element-deps                      |  2 +-
 elements/mariadb/install.d/10-mariadb              | 36 ----------------------
 elements/mariadb/install.d/10-mariadb-packages     | 13 ++++++++
 .../os-refresh-config/post-configure.d/40-mariadb  |  4 ---
 .../pre-configure.d/50-mariadb-socket              | 10 ------
 18 files changed, 89 insertions(+), 51 deletions(-)
 create mode 100644 elements/mariadb-common/README.md
 create mode 100644 elements/mariadb-common/element-deps
 create mode 100755 elements/mariadb-common/install.d/11-mariadb
 create mode 100755 elements/mariadb-common/os-refresh-config/post-configure.d/40-mariadb
 create mode 100755 elements/mariadb-common/os-refresh-config/pre-configure.d/50-mariadb-socket
 create mode 100644 elements/mariadb-dev-rdo/README.md
 create mode 100644 elements/mariadb-dev-rdo/element-deps
 create mode 100644 elements/mariadb-dev-rdo/element-provides
 create mode 100755 elements/mariadb-dev-rdo/install.d/03-mariadb-dev
 create mode 100644 elements/mariadb-rdo/README.md
 create mode 100644 elements/mariadb-rdo/element-deps
 create mode 100644 elements/mariadb-rdo/element-provides
 create mode 100755 elements/mariadb-rdo/install.d/10-mariadb-packages
 delete mode 100755 elements/mariadb/install.d/10-mariadb
 create mode 100755 elements/mariadb/install.d/10-mariadb-packages
 delete mode 100755 elements/mariadb/os-refresh-config/post-configure.d/40-mariadb
 delete mode 100755 elements/mariadb/os-refresh-config/pre-configure.d/50-mariadb-socket

diff --git a/elements/mariadb-common/README.md b/elements/mariadb-common/README.md
new file mode 100644
index 0000000..50a9d23
--- /dev/null
+++ b/elements/mariadb-common/README.md
@@ -0,0 +1,2 @@
+This element contains code common for mariadb installations, make sure
+you include one of mariadb or mariadb-rdo when including this element.
diff --git a/elements/mariadb-common/element-deps b/elements/mariadb-common/element-deps
new file mode 100644
index 0000000..b51cf12
--- /dev/null
+++ b/elements/mariadb-common/element-deps
@@ -0,0 +1 @@
+mysql-common
diff --git a/elements/mariadb-common/install.d/11-mariadb b/elements/mariadb-common/install.d/11-mariadb
new file mode 100755
index 0000000..cf14849
--- /dev/null
+++ b/elements/mariadb-common/install.d/11-mariadb
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Install controller base requiered packages
+
+set -eux
+
+# Templates write the configs into /mnt/state. However, MySQL makes it very
+# difficult not to use this as the directory for configs.
+rm -rf /etc/mysql
+ln -s /mnt/state/etc/mysql /etc/mysql
+if [ -e /etc/apparmor.d/usr.sbin.mysqld ] ; then
+    sed -i -e 's,/var/lib/mysql/,/mnt/state/var/lib/mysql/,g' /etc/apparmor.d/usr.sbin.mysqld
+    sed -i -e 's,/var/log/mysql/,/mnt/state/var/log/mysql/,g' /etc/apparmor.d/usr.sbin.mysqld
+    sed -i -e 's,/etc/mysql/,/mnt/state/etc/mysql/,g' /etc/apparmor.d/usr.sbin.mysqld
+fi
+if [ -e /etc/init/mysql.conf ]; then
+    sed -i -e 's,/var/lib/mysql/,/mnt/state/var/lib/mysql/,g' /etc/init/mysql.conf
+fi
+# Fedora/RHEL install /etc/my.cnf but we do not want any unmanaged configs
+rm -f /etc/my.cnf
+# On openSUSE /var/lib/mysql is not part of the mariadb packages.
+[ -d /var/lib/mysql ] || install -d -o mysql -g root -m 0700 /var/lib/mysql
+register-state-path /var/lib/mysql
+# We need to setup the directory with appropriate permissions and then
+# the first time we boot a particular state partition we rsync this in.
+[ -d /var/log/mysql ] || install -d -o root -g mysql -m 0775 /var/log/mysql
+register-state-path /var/log/mysql
diff --git a/elements/mariadb-common/os-refresh-config/post-configure.d/40-mariadb b/elements/mariadb-common/os-refresh-config/post-configure.d/40-mariadb
new file mode 100755
index 0000000..948bc26
--- /dev/null
+++ b/elements/mariadb-common/os-refresh-config/post-configure.d/40-mariadb
@@ -0,0 +1,4 @@
+#!/bin/bash
+set -eu
+
+os-svc-restart -n mysql
diff --git a/elements/mariadb-common/os-refresh-config/pre-configure.d/50-mariadb-socket b/elements/mariadb-common/os-refresh-config/pre-configure.d/50-mariadb-socket
new file mode 100755
index 0000000..f6b800c
--- /dev/null
+++ b/elements/mariadb-common/os-refresh-config/pre-configure.d/50-mariadb-socket
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -eu
+
+# mariadb clients (from mariadb.org) expect the socket at
+# /var/lib/mysql/mysql.sock
+#
+# mysql server is started from 51-init-openstack (reset-db script) - we need to
+# create symlink to make sure reset-db will not fail
+[ -d /var/lib/mysql ] || mkdir -p /var/lib/mysql
+ln -sf /var/run/mysqld/mysqld.sock /var/lib/mysql/mysql.sock
diff --git a/elements/mariadb-dev-rdo/README.md b/elements/mariadb-dev-rdo/README.md
new file mode 100644
index 0000000..6f9afc1
--- /dev/null
+++ b/elements/mariadb-dev-rdo/README.md
@@ -0,0 +1,4 @@
+Installs mariadb devel package from RDO repository
+
+This is a separate element because different devel packages are installed for
+mysql and for mariadb.
diff --git a/elements/mariadb-dev-rdo/element-deps b/elements/mariadb-dev-rdo/element-deps
new file mode 100644
index 0000000..c77dd9b
--- /dev/null
+++ b/elements/mariadb-dev-rdo/element-deps
@@ -0,0 +1 @@
+fedora-rdo-icehouse-repository
diff --git a/elements/mariadb-dev-rdo/element-provides b/elements/mariadb-dev-rdo/element-provides
new file mode 100644
index 0000000..063f2a8
--- /dev/null
+++ b/elements/mariadb-dev-rdo/element-provides
@@ -0,0 +1 @@
+mysql-dev
diff --git a/elements/mariadb-dev-rdo/install.d/03-mariadb-dev b/elements/mariadb-dev-rdo/install.d/03-mariadb-dev
new file mode 100755
index 0000000..84937ce
--- /dev/null
+++ b/elements/mariadb-dev-rdo/install.d/03-mariadb-dev
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+set -eux
+
+install-packages libmariadb-dev
diff --git a/elements/mariadb-rdo/README.md b/elements/mariadb-rdo/README.md
new file mode 100644
index 0000000..bc29728
--- /dev/null
+++ b/elements/mariadb-rdo/README.md
@@ -0,0 +1,4 @@
+Installs MariaDB with galera from RDO repository
+
+This element is similar to mariadb repository but installs packages
+from RDO repository.
diff --git a/elements/mariadb-rdo/element-deps b/elements/mariadb-rdo/element-deps
new file mode 100644
index 0000000..28f3ac7
--- /dev/null
+++ b/elements/mariadb-rdo/element-deps
@@ -0,0 +1,2 @@
+mariadb-dev-rdo
+mariadb-common
diff --git a/elements/mariadb-rdo/element-provides b/elements/mariadb-rdo/element-provides
new file mode 100644
index 0000000..0eaebf1
--- /dev/null
+++ b/elements/mariadb-rdo/element-provides
@@ -0,0 +1 @@
+mysql
diff --git a/elements/mariadb-rdo/install.d/10-mariadb-packages b/elements/mariadb-rdo/install.d/10-mariadb-packages
new file mode 100755
index 0000000..88257f6
--- /dev/null
+++ b/elements/mariadb-rdo/install.d/10-mariadb-packages
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Install controller base requiered packages
+
+set -eux
+
+if [ "$(dib-init-system)" = "upstart" ] ; then
+    install $(dirname $0)/mysql-set-server-id.upstart /etc/init/mysql-set-server-id.conf
+else
+    echo WARNING: server-id will not be set on systems that boot this image!
+fi
+
+install-packages sysstat mytop python-mysqldb mariadb-rdo-galera-server mariadb-galera galera
diff --git a/elements/mariadb/element-deps b/elements/mariadb/element-deps
index 6de744a..dfdb858 100644
--- a/elements/mariadb/element-deps
+++ b/elements/mariadb/element-deps
@@ -1,2 +1,2 @@
 mariadb-dev
-mysql-common
+mariadb-common
diff --git a/elements/mariadb/install.d/10-mariadb b/elements/mariadb/install.d/10-mariadb
deleted file mode 100755
index 19e613a..0000000
--- a/elements/mariadb/install.d/10-mariadb
+++ /dev/null
@@ -1,36 +0,0 @@
-#!/bin/bash
-
-# Install controller base requiered packages
-
-set -e
-set -o xtrace
-
-if [ "$(dib-init-system)" = "upstart" ] ; then
-    install $(dirname $0)/mysql-set-server-id.upstart /etc/init/mysql-set-server-id.conf
-else
-    echo WARNING: server-id will not be set on systems that boot this image!
-fi
-
-install-packages sysstat mytop python-mysqldb mariadb-galera-server mariadb-client galera
-
-# Templates write the configs into /mnt/state. However, MySQL makes it very
-# difficult not to use this as the directory for configs.
-rm -rf /etc/mysql
-ln -s /mnt/state/etc/mysql /etc/mysql
-if [ -e /etc/apparmor.d/usr.sbin.mysqld ] ; then
-    sed -i -e 's,/var/lib/mysql/,/mnt/state/var/lib/mysql/,g' /etc/apparmor.d/usr.sbin.mysqld
-    sed -i -e 's,/var/log/mysql/,/mnt/state/var/log/mysql/,g' /etc/apparmor.d/usr.sbin.mysqld
-    sed -i -e 's,/etc/mysql/,/mnt/state/etc/mysql/,g' /etc/apparmor.d/usr.sbin.mysqld
-fi
-if [ -e /etc/init/mysql.conf ]; then
-    sed -i -e 's,/var/lib/mysql/,/mnt/state/var/lib/mysql/,g' /etc/init/mysql.conf
-fi
-# Fedora/RHEL install /etc/my.cnf but we do not want any unmanaged configs
-rm -f /etc/my.cnf
-# On openSUSE /var/lib/mysql is not part of the mariadb packages.
-[ -d /var/lib/mysql ] || install -d -o mysql -g root -m 0700 /var/lib/mysql
-register-state-path /var/lib/mysql
-# We need to setup the directory with appropriate permissions and then
-# the first time we boot a particular state partition we rsync this in.
-[ -d /var/log/mysql ] || install -d -o root -g mysql -m 0775 /var/log/mysql
-register-state-path /var/log/mysql
diff --git a/elements/mariadb/install.d/10-mariadb-packages b/elements/mariadb/install.d/10-mariadb-packages
new file mode 100755
index 0000000..e33e8c9
--- /dev/null
+++ b/elements/mariadb/install.d/10-mariadb-packages
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Install controller base requiered packages
+
+set -eux
+
+if [ "$(dib-init-system)" = "upstart" ] ; then
+    install $(dirname $0)/mysql-set-server-id.upstart /etc/init/mysql-set-server-id.conf
+else
+    echo WARNING: server-id will not be set on systems that boot this image!
+fi
+
+install-packages sysstat mytop python-mysqldb mariadb-galera-server mariadb-client galera
diff --git a/elements/mariadb/os-refresh-config/post-configure.d/40-mariadb b/elements/mariadb/os-refresh-config/post-configure.d/40-mariadb
deleted file mode 100755
index 948bc26..0000000
--- a/elements/mariadb/os-refresh-config/post-configure.d/40-mariadb
+++ /dev/null
@@ -1,4 +0,0 @@
-#!/bin/bash
-set -eu
-
-os-svc-restart -n mysql
diff --git a/elements/mariadb/os-refresh-config/pre-configure.d/50-mariadb-socket b/elements/mariadb/os-refresh-config/pre-configure.d/50-mariadb-socket
deleted file mode 100755
index f6b800c..0000000
--- a/elements/mariadb/os-refresh-config/pre-configure.d/50-mariadb-socket
+++ /dev/null
@@ -1,10 +0,0 @@
-#!/bin/bash
-set -eu
-
-# mariadb clients (from mariadb.org) expect the socket at
-# /var/lib/mysql/mysql.sock
-#
-# mysql server is started from 51-init-openstack (reset-db script) - we need to
-# create symlink to make sure reset-db will not fail
-[ -d /var/lib/mysql ] || mkdir -p /var/lib/mysql
-ln -sf /var/run/mysqld/mysqld.sock /var/lib/mysql/mysql.sock
-- 
1.9.0

